src/components/CheckList/CheckFilterGroup.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { Button, Icon, useStyles } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { GrafanaTheme } from '@grafana/data';
+import { defaultFilters, CheckFilters } from './CheckList';
+
+const groupStyles = (theme: GrafanaTheme) => ({
+  container: css`
+    position: relative;
+  `,
+  dropdown: css`
+    position: absolute;
+    background-color: rgb(24, 27, 31);
+    border: 1px solid rgba(204, 204, 220, 0.15);
+    border-radius: 2px;
+    z-index: 100;
+    padding: 20px;
+    margin-top: 5px;
+    -webkit-box-shadow: 5px 5px 14px -3px rgba(0, 0, 0, 0.67);
+    box-shadow: 5px 5px 14px -3px rgba(0, 0, 0, 0.67);
+    right: 8px;
+    width: 500px;
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+  `,
+  horizontalGroup: css`
+    margin-top: 20px;
+    margin-bottom: 0px;
+    display: flex;
+    justify-content: flex-end;
+    align-self: flex-end;
+  `,
+  marginRightSmall: css`
+    margin-right: ${theme.spacing.sm};
+  `,
+});
+
+interface Props {
+  children: JSX.Element[] | JSX.Element;
+  onReset: () => void;
+  filters: CheckFilters;
+}
+
+const CheckFilterGroup = ({ children, onReset, filters }: Props) => {
+  const [openFilters, setOpenFilters] = useState(false);
+  const [activeFilters, setActiveFilters] = useState(0);
+  const styles = useStyles(groupStyles);
+
+  const handleFilterOpen = useCallback(() => {
+    setOpenFilters(!openFilters);
+  }, [openFilters]);
+
+  useEffect(() => {
+    let active = 0;
+    // Count which filters have been applied
+    Object.keys(filters).map((key) => {
```

Comment:
philisophical nit: this could be a forEach or a reduce since you don't need the array produced by mapping

Reply:
Yes, yes it could